### DISCUSSION TITLE
Wsparcie dla wykazów DAB+

### DISCUSSION THREAD

**Discussion by SzymonDX** [TomaszGasior/RadioLista-v3#146] [2. Pomysły Ideas]
Proponuję dostosowanie Radiolisty pod wykazy cyfrowego radia (DAB+).

Aktualnie brakuje kolumny ‘kanał’ (w DABie kanał odbioru podaje się na równi z częstotliwością). 
Myślę, że warto zrezygnować z zaokrąglania liczbowego przy częstotliwości (aktualnie z częstotliwości 202,928 MHz robi się 202,95 MHz). 
Dodatkowo przydałoby się pole ‘lista kanałów’, aby wpisać kanały dostępne w danym multipleksie DAB+ - w sumie mogłoby to wyglądać jak obecnie kolumna z RDSem.

Uważam, że dobrym pomysłem jest stworzenie wyboru typu wykazu – domyślnie FM, z możliwością wyboru np. DAB+ / AM / TV. A to dlatego, że część kolumn między tymi pasmami będzie się różnić, stąd przy późniejszej rozbudowie wykazu wrzucanie wszystkich kolumn do jednego worka nie ma raczej sensu. Przejrzystość interfejsu również na tym zyska.

**Comment by TomaszGasior** [comment]
Czy mógłbym cię prosić o podanie linków do przykładowych wykazów DAB+, abym mógł sobie zobaczyć jak one wyglądają?

**Comment by HyperDX** [comment]
@TomaszGasior Ciężko jest znaleźć tropo-logi DXerów w paśmie DAB+, a troszkę już szukałem. Jak na razie jeśli chodzi o wykaz wszystkich emisji to jest na przykład ten: http://radiopolska.pl/wykaz/dab, który mniej więcej pokazuje jak powinien wyglądać rozkład elementów w wykazie dla tego pasma.

**Comment by TomaszGasior** [comment]
@HyperDX  

Wydaje mi się, że źle umieszczasz odnośniki w formacie Markdown i tam, gdzie powinien być tekst linka, wklejasz adres, a tam gdzie ma być adres, zostawiasz domyślne — dlatego twoich linków nigdy nie da się kliknąć. Możesz je po prostu luzem wkleić do treści: wtedy automatycznie będą klikalne.

---

@SzymonDX 

> Myślę, że warto zrezygnować z zaokrąglania liczbowego przy częstotliwości (aktualnie z częstotliwości 202,928 MHz robi się 202,95 MHz).

Możemy zastosować rozwiązanie użyte w polu mocy nadajnika: przy wyświetlaniu wartości są używane co najmniej dwa miejsca po przecinku, ale opcjonalnie jest możliwość wprowadzenia trzeciego miejsca po przecinku. Będzie takie coś pasowało? Jeśli tak, bardzo proszę o założenie oddzielnego zgłoszenia dla tego problemu.

> Dodatkowo przydałoby się pole ‘lista kanałów’, aby wpisać kanały dostępne w danym multipleksie DAB+ - w sumie mogłoby to wyglądać jak obecnie kolumna z RDSem.

Mamy dwa podejścia:
- Podejście A: wykazy zawierające stacje radiowe, których cechą jest m.in. to, na jakim multipleksie dana stacja się znajduje.
- Podejście B: wykazy zawierające multipleksy, których cechą jest m.in. jakie stacje radiowe zawiera dany multipleks.

Mam wrażenie, że obydwa podejścia mają sens i mogą być pożyteczne dla użytkowników. Można je połączyć w jednym serwisie, nie wykluczają się, natomiast trzeba to zrobić z głową. Od samego początku w 2012 roku design RadioListy opiera się na tym, że w wykazach znajdują się **stacje radiowe** (lub z delikatnym naciąganiem, także TV). Gdybym teraz tak po prostu do pola obiektu „stacja radiowa” dodał pole „lista kanałów” to będzie to kompletnie bez sensu z perspektywy designu serwisu: no bo jak stacja może zawierać stacje? Gdy natomiast do obiektu „stacja radiowa” dodam pole „multipleks” to będzie to miało sens.

To, co chce powiedzieć to to, że obydwa podejścia mają sens i są wykonalne, natomiast podejście A „dodawajmy każdą stację oddzielnie, wpisując na którym jest MUX-ie” (z możliwością posortowania po MUX-ach) będzie bardziej pasować do obecnej konstrukcji serwisu oraz będzie po prostu prostsze i szybsze dla mnie do zrobienia. Przekonstruowanie całej architektury serwisu tak, by dało się zrobić oddzielny typ wykazu z MUX-ami, a nie stacjami, wymagać będzie więcej pracy. Nie wiem, czy będę w stanie czasowo zrealizować coś takiego. 

Poza tym warto jest po prostu konceptualnie oddzielić te podejścia. Dlatego bardzo proszę, wskaż mi tutaj na którym z tych dwóch podejść chciałbyś żebyśmy się skupili, a dla tego drugiego możemy ew. zrobić oddzielny wątek. Możemy np. zrobić tak, że w tym wątku skupimy się na podejściu A, natomiast możemy zrobić oddzielny wątek dla podejścia B, w którym razem: ty z perspektywy usera, ja z perspektywy programisty, opiszemy jak to podejście należy zrealizować i jeśli ktoś w przyszłości będzie chciał poświęcić na to czas, będzie mógł posłużyć się tymi pomysłami.

RadioLista jest open source właśnie po to, żeby oprócz mnie serwisem mogły zajmować się też inne osoby, które mają na to ochotę, dlatego takie rozpisanie i podzielenie pomysłów ma dużo sensu nawet, jeśli nie mogę obiecać, że ja coś zrealizuję we własnym czasie. Tak działa open source. 

(Tak przy okazji. Te zgłoszenia, które „przypisuję” do siebie tzn. we właściwościach zgłoszenia po prawej w polu „Assignees” widać mój nick, to te zgłoszenia, w których deklaruję, że postaram się je zrealizować w najbliższym czasie i zostawiam je dla siebie. Te, które takiej „przypinki” nie mają i/lub mają etykietkę „help needed” są zostawione dla innych zainteresowanych — choć oczywiście w dalszej przyszłości mogę się sam do nich przypisać z powrotem. :) )

> Uważam, że dobrym pomysłem jest stworzenie wyboru typu wykazu – domyślnie FM, z możliwością wyboru np. DAB+ / AM / TV

Będę bardzo wdzięczny, jeśli założysz kolejny oddzielny wątek, w którym opiszesz, jak to dokładnie mogłoby wyglądać. Na przykład wypiszesz wszystkie rodzaje wykazów i dokładnie czym mają się różnic z twojej perspektywy, z perspektywy użytkownika. :)

**Comment by TomaszGasior** [comment]
Dla typu wykazów założylem zgloszenie #120 
Dla kanału założyłem zgłoszenie #122 

@SzymonDX @HyperDX Proszę was o pomoc. Będę wdzięczny za opisanie w #122 jaka jest dokładna specyfikacja kolumny kanału — czy zawiera cyfry, liczby, jaki zakres, ile znaków. Proszę też o pomoc w zadaniu #121: jakie byście widzieli predefiniowane „zestawy” kolumn wykazu pod konkretne zastosowania?

**Comment by TomaszGasior** [comment]
To zgłoszenie zamykam, jako że spełniło swoje zadanie. @SzymonDX Możesz zasubskrybować zalinkowane wyżej taski, żeby być na bieżąco z postępem.

**Comment by TomaszGasior** [comment]
@HyperDX  @SzymonDX 

Przekonwertowałem zamknięte zgłoszenie do wątku na forum dyskusyjnym ze względu na nowe założenie polegające na tym, że sekcja „issues” będzie przeznaczona głównie dla mnie i do opisów technicznych, a sekcja forum dyskusyjnego — dla użytkowników. 

Będę zajmować się w najbliższym czasie tematem dodania nowej kolumny kanału do wykazu, aby ułatwić zarządzanie wykazami DAB+ takimi jak https://radiolista.pl/wykaz/248. Gdy będzie gotowe, wystawię do przetestowania wam wersję demonstracyjną.

**Comment by TomaszGasior** [comment]
@HyperDX  @SzymonDX 

Moi drodzy, mamy postęp. :tada:  Co prawda po dwóch latach. :cry:  Ale jest.

- Trzy miejsca po przecinku w częstotliwości. #147 
- Kolumna „Multipleks cyfrowy” („Mux”). #149 
- Kolumna „Kanał DAB” („Kan.”). #122 
- Automatyczne uzupełnienie częstotliwości przy zmianie kanału DAB.

Te wszystkie rzeczy czekają teraz na was w **instancji testowej RadioListy** na dev.radiolista.pl. Rzućcie okiem i sprawdźcie, czy wszystko działa zgodnie z oczekiwaniami. Jeśli wszystko jest OK, wrzucę aktualizację na główną instancję serwisu.

**Reply by HyperDX** [reply]
Właśnie testuje i mam uwagę. Dokładnie odnośnie nazewnictwa kolumny "Mux". Przez wiele lat utarła się natomiast pisownia tego skrótu jako "MUX" z dużych liter. 

Poza tym jest wszystko tak jak trzeba. Auto uzupełnianie działa bez zarzutu. 

Jeszcze jedna nurtująca mnie kwestia, ponownie odnosząca się do kolumny "MUX", a chodzi o jej zawartość. Mamy już definiowaną kolumnę "Nazwa", do której można po prostu wpisać nazwę danego multipleksu. Jeśli chodzi o MUX to może tutaj zamieszczać zawartość samego multipleksu jeśli chodzi o programy w nim zamieszczone (forma może jako rozwijany pop-up, coś w stylu RDS). 

Lub po prostu do kolumny "MUX" wpisać nazwę danego multipleksu (ukrywając kolumnę "Nazwa" przy okazji), a programy, które w nim są dać do jakiejś nowej kolumny lub komentarza. @SzymonDX przedstaw swoją wizję, bo będzie ona tu bardzo pomocna, a pewnie niektórzy będą chcieli dodać informację jakie stacje radiowe znajdują się w danym MUX-ie :).

**Reply by TomaszGasior** [reply]
Nazwę poprawię. Co do sposobu użycia kolumn „MUX” i „Nazwa” intencja jest taka, by w polu „Nazwa” wpisać nazwę stacji radiowej, a w polu „Mux” — multipleks tejże stacji. Napisałem już dawno temu (tutaj wyżej go widać) komentarz https://github.com/TomaszGasior/RadioLista-v3/discussions/146#discussioncomment-812803:

> Mamy dwa podejścia:
>
>    - Podejście A: wykazy zawierające stacje radiowe, których cechą jest m.in. to, na jakim multipleksie dana stacja się znajduje.
>    - Podejście B: wykazy zawierające multipleksy, których cechą jest m.in. jakie stacje radiowe zawiera dany multipleks.
>
> Mam wrażenie, że obydwa podejścia mają sens i mogą być pożyteczne dla użytkowników. Można je połączyć w jednym serwisie, nie wykluczają się, natomiast trzeba to zrobić z głową. Od samego początku w 2012 roku design RadioListy opiera się na tym, że w wykazach znajdują się stacje radiowe (lub z delikatnym naciąganiem, także TV). Gdybym teraz tak po prostu do pola obiektu „stacja radiowa” dodał pole „lista kanałów” to będzie to kompletnie bez sensu z perspektywy designu serwisu: no bo jak stacja może zawierać stacje? Gdy natomiast do obiektu „stacja radiowa” dodam pole „multipleks” to będzie to miało sens.
>
> To, co chce powiedzieć to to, że obydwa podejścia mają sens i są wykonalne, natomiast podejście A „dodawajmy każdą stację oddzielnie, wpisując na którym jest MUX-ie” (z możliwością posortowania po MUX-ach) będzie bardziej pasować do obecnej konstrukcji serwisu oraz będzie po prostu prostsze i szybsze dla mnie do zrobienia. Przekonstruowanie całej architektury serwisu tak, by dało się zrobić oddzielny typ wykazu z MUX-ami, a nie stacjami, wymagać będzie więcej pracy. Nie wiem, czy będę w stanie czasowo zrealizować coś takiego.

**Reply by HyperDX** [reply]
Tu się rodzi pytanie. Podczas odbioru multipleksu DAB/DAB+ nie odbieramy tylko jednej stacji, ale całą paczkę rozgłośni. Jeśli pojedyncze dodawanie każdej stacji, która należy do danego multipleksu nie zaburzy samej czytelności wykazu to jak najbardziej jestem za twoim tokiem myślenia. :)

**Reply by SzymonDX** [reply]
Nowe funkcje będą przydatne, z góry dzięki! 
Miałem taki zamysł, by kolumna tutaj nazwana "MUX" była zawartością programową ("Lista kanałów" albo coś w ten deseń) - czyli po prostu tym, co dana paczka kanałów zawiera. 
Oczywiście wyświetlanie przykładowo 15 różnych nazw stacji byłoby średnio wygodne, więc powinno to się odbywać w formie okienka / rozwijanej listy - po kliknięciu czy najechaniu na ten fragment wykazu; mogłoby to działać identycznie, jak to ma miejsce z wyświetlaniem szczegółowych danych RDSu.

**Reply by TomaszGasior** [reply]
Od 2012 roku RadioLista była kontruowana w taki sposób, że elementami wykazu są stacje radiowe (nie multipleksy lub inne obiekty).  Powstający od [sierpnia 2018 roku](https://github.com/TomaszGasior/RadioLista-v3/commit/d3addab407c154cb59eeb22e10a363ea8d2fe557) kod również przyjmuje takie założenie. Dlatego właśnie w stajach radiowych można ustawić np. rodzaj nadawnego programu czy numer w odbiorniku. Nie mogę dodać ot tak kolumny „lista kanałów”, bo nie będzie to maiło sensu ani z perspektywy dotychczas powstałego kodu, ani z perspektywy szeroko pojętego user experience aplikacji. Z perspektywy obecnej architektury serwisu fakt, że na jednym multipleksie nadaje się wiele stacji jest szczegółem dookoła stacji tak samo jak to, że wiele stacji nadaje się np. z jednego masztu nadawczego. 

Stworzenie oddzielnego rodzaju wykazu z multipleksami zamiast stacjami radiowymi jest możliwe — skłamałbym, gdybym powiedział inaczej. :) Jednak wymagałoby to bardzo dużo pracy: obsługi dwóch różnych metod wybierania elementów wykazu, stworzenia i utrzymania dodatkowych ekranów/formularzy edycji multipleksów na równi ze stacjami.  Ze względu na przewidywany koszt czasowy na razie zostanie tak, jak jest.

Dziękuję za wasze uwagi i przepraszam za długi czas oczekiwania na implementację tej dosyć prostej funkcji. :)  Efektem naszej wspólnej pracy jest https://github.com/TomaszGasior/RadioLista-v3/discussions/150

**Comment by TomaszGasior** [comment]
@HyperDX  @SzymonDX Zapraszam do nowo otwartej dyskusji związanej z tym tematem: https://github.com/TomaszGasior/RadioLista-v3/discussions/211
